pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumBGPPeerConfigList is a list of CiliumBGPPeer objects.
+type CiliumBGPPeerConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumBGPPeer.
+	Items []CiliumBGPPeerConfig `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgppeerconfig",path="ciliumbgppeerconfigs",scope="Cluster",shortName={bgppeer}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+type CiliumBGPPeerConfig struct {
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec is the specification of the desired behavior of the CiliumBGPPeerConfig.
+	Spec CiliumBGPPeerConfigSpec `json:"spec,omitempty"`
+}
+
+type CiliumBGPPeerConfigSpec struct {
+	// Transport defines the BGP transport parameters for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Transport *CiliumBGPTransport `json:"transport,omitempty"`
+
+	// Timers defines the BGP timers for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Timers *CiliumBGPTimers `json:"timers,omitempty"`
+
+	// AuthSecretRef is the name of the secret to use to fetch a TCP
+	// authentication password for this peer.
+	// +kubebuilder:validation:Optional
+	AuthSecretRef *string `json:"authSecretRef,omitempty"`
+
+	// GracefulRestart defines graceful restart parameters which are negotiated
+	// with this peer. If empty / nil, the graceful restart capability is disabled.
+	//
+	// +kubebuilder:validation:Optional
+	GracefulRestart *CiliumBGPNeighborGracefulRestart `json:"gracefulRestart,omitempty"`
+
+	// EBGPMultihopTTL controls the multi-hop feature for eBGP peers.
+	// Its value defines the Time To Live (TTL) value used in BGP
+	// packets sent to the peer.
+	// The value 1 implies that eBGP multi-hop feature is disabled
+	// (only a single hop is allowed).
```

Comment:
`1` and unspecified mean the same behavior, e.g. "disabled". We should consider simplifying the API by simply saying "When unset, EBGP multihop is disabled. This field is ignored for iBGP neighbors.".